python/pysrc/kaskada/timestream/__init__.py

```diff
@@ -0,0 +1,122 @@
+"""Defines Kaskada Timestreams."""
+
+from __future__ import annotations
+
+# this must be imported first to avoid circular imports
+from ._timestream import Timestream, Arg, LiteralValue, record, Literal
+
+# after Timestream is initialized, add all the methods
+from ._aggregation import collect, count, count_if, first, last, max, mean, min, stddev, sum, variance
+from ._arithmetic import add, add__, addr__, ceil, clamp, div, div__, divr__, exp, floor, greatest, least, mul, mul__, mulr__, neg, powf, round, sqrt, sub, sub__, subr__
+from ._collection import getitem__, flatten, index, length, union
+from ._comparison import eq, ge, ge__, gt, gt__, le, le__, lt, lt__, ne, is_not_null, is_null
+from ._execution import preview, to_pandas, run_iter, write
+from ._grouping import lookup, with_key
+from ._logical import and_, or_, not_
+from ._misc import cast, coalesce, else_, filter, hash, if_, lag, null_if, pipe
+from ._records import col, extend, _record, remove, select
+from ._string import lower, upper
+from ._time import shift_by, shift_to, shift_until, time, seconds_since, seconds_since_previous
+
+# aggregation
+Timestream.collect = collect
```

Comment:
This seems pretty messy, and I'm not sure how idiomatic it is.

Looking around, it seems it's not particularly suggested to split up a single class like this. But if we need to, https://stackoverflow.com/questions/47561840/how-can-i-separate-the-functions-of-a-class-into-multiple-files may be a bit cleaner:

```
class Timestream:
   def __init__(self,whatever):
     ...

     # Imported methods
     from ._arithmetic import add, sub, mul, div
```

The benefit being that you don't need to separately import them and then set them on the class -- you can just import them directly into the class.

Reply:
I tried this first, but it doesn't work due to circular references.

Reply:
I realize this is messy, and maybe we shouldn't do this, but I also really dislike the `_timestreams.py` file being so huge and unorganized.

Reply:
Hmm. There are ways to avoid that -- I can take a look tomorrow. Usually, the trick is to not import them via `__init__.py` and rather import them from the file they are defined in. Basically, `__init__.py` is just used to create the public / external modules.

See for instance https://medium.com/brexeng/avoiding-circular-imports-in-python-7c35ec8145ed